management/user.go

```diff
@@ -343,3 +343,19 @@ func (m *UserManager) Blocks(id string, opts ...RequestOption) ([]*UserBlock, er
 func (m *UserManager) Unblock(id string, opts ...RequestOption) error {
 	return m.Request("DELETE", m.URI("user-blocks", id), nil, opts...)
 }
+
+// Link links two user accounts together forming a primary and secondary relationship.
+//
+// See: https://auth0.com/docs/api/management/v2#!/Users/post_identities
+func (m *UserManager) Link(id string, il *IdentityLink, opts ...RequestOption) (err error) {
+	return m.Request("POST", m.URI("users", id, "identities"), il, opts...)
+}
+
+// IdentityLink contains the data needed for linking an identity to a given user.
+//
+// Note: connection_id param is needed just to identify a particular database connection for the 'auth0' provider.
+type IdentityLink struct {
+	ConnectionID *string `json:"connection_id,omitempty"`
+	UserID       string  `json:"user_id"`
+	Provider     string  `json:"provider"`
```

Comment:
Can we include the comments from the api docs?

```
// Identity provider of the secondary user account being linked.
Provider *string ...
```

`UserID` and `Provider` should also be pointers.

We're missing `link_with` according to the api docs: JWT for the secondary account being linked. If sending this parameter, provider, user_id, and connection_id must not be sent.

Reply:
For the `link_with` param, it actually requires a JWT from the primary account in the `authorization` header. I think this library does not support this specific authentication mechanism, that's why I didn't add the param; Not adding the param made the `UserID` and `Provider` params required (so non-pointers).
WDYT?

Reply:
`UserID` and `Provider` should be pointers anyway as to preserve the conventions used throughout the SDK.

Reply:
I'd also include `link_with` just in case. Perhaps someone might want to use it together with other tools?